src/extend-helpers.js

```diff
@@ -1,34 +1,43 @@
-'use strict';
+function deprecate(helper) {
+	return function (...args) {
+		if (process.env.NODE_ENV !== 'production') {
+			// eslint-disable-next-line no-console
+			console.warn(`The Handlebars helper ${helper.name} has been deprecated.`);
+		}
+
+		return Reflect.apply(helper, this, args);
```

Comment:
I haven't seen this pattern before! I'm not sure what `args` is or why we need to use `Reflect.apply` rather than calling the required helper. Is it because the argument we're passing to deprecate is a `require` statement? 🤔

Reply:
This is equivalent to `helper.apply(this, args)` - in either case the important point is to preserve `this` as Handlebars calls helpers with this context.

I'm capturing all of the function arguments with a rest parameter which is convenient for two reasons:

1. I don't know how many arguments there might be
2. It captures all arguments into an array. I could also use the magic `arguments` object but then I'd have to make it into an array myself.